libs/wingc/src/type_check.rs

```diff
@@ -134,6 +134,8 @@ pub enum Type {
 	Duration,
 	Boolean,
 	Void,
+	Json,
+	MutJson,
```

Comment:
Now that we have these types we need to type check them when they appear in literals. Basically we need to verify that all fields within a `Json` literal are valid json types. So something like this won't work:
```
let b = new cloud.Bucket();
let arr = Json [b,b,b];
```
You'll probably need to add code for this in `type_check_exp`

Reply:
hmm, yea right now the acceptable json types are pretty much everything, I want to trim this down but curious how that would look since we would have to allow for sub arrays within an array and without maybe like a serializable interface not sure how we could limit the allowed types of arrays. IE. `[ "a", 2, [b, b, b]]`